ui/lib/core/addon/components/search-select.js

```diff
@@ -1,99 +1,110 @@
- * @param {string | Array} inputValue -  A comma-separated string or an array of strings -- array of ids for models.
- * @param {string} label - Label for this form field
- * @param {string} fallbackComponent - name of component to be rendered if the API call 403s
- * @param {string} [backend] - name of the backend if the query for options needs additional information (eg. secret backend)
+ * @param {array} [inputValue] - Array of strings corresponding to the input's initial value, e.g. an array of model ids that on edit will appear as selected items below the input
  * @param {boolean} [disallowNewItems=false] - Controls whether or not the user can add a new item if none found
- * @param {boolean} [passObject=false] - When true, the onChange callback returns an array of objects with id (string) and isNew (boolean) (instead of an array of id strings)
- * @param {array} [objectKeys=null] - Array of values that correlate to model attrs. When passObject=true, objectKeys are added to the passed object. NOTE: make 'id' as the first element in objectKeys if you do not want to override the default of 'id'
- * @param {string} [helpText] - Text to be displayed in the info tooltip for this form field
- * @param {number} [selectLimit] - A number that sets the limit to how many select options they can choose
+ * @param {boolean} [shouldRenderName=false] - By default an item's id renders in the dropdown, `true` displays the name with its id in smaller text beside it *NOTE: the boolean flips automatically with 'identity' models or if this.idKey !== 'id'
+ * @param {array} [parentManageSelected] - Array of selected items if the parent is keeping track of selections, see mfa-login-enforcement-form.js
+ * @param {boolean} [passObject=false] - When true, the onChange callback returns an array of objects with id (string) and isNew (boolean) (and any params from objectKeys). By default - onChange returns an array of id strings. 
+ * @param {array} [objectKeys=null] - Array of values that correlate to model attrs. When passObject=true, objectKeys are added to the passed, selected object. *NOTE: make 'id' as the first element in objectKeys if you do not want to override the default of 'id'
+ * @param {number} [selectLimit] - Sets select limit
+ 
+// * query params for dropdown items
+ * @param {Array} models - An array of model types to fetch from the API.
+ * @param {string} [backend] - name of the backend if the query for options needs additional information (eg. secret backend)
+ * @param {object} [queryObject] - object passed as query options to this.store.query(). NOTE: will override @backend
+ 
+ // * template only/display args
+ * @param {string} id - The name of the form field
+ * @param {string} [label] - Label for this form field, also used to generate `Add new ${singularize(label}` copy when adding a nonexisting option
```

Comment:
Not sure we should tie the label so closely to the item noun as it's used in the create new option. More and more of our labels are descriptive or instructive rather than just what the thing is, so maybe we use this by default but provide an optional arg where it can be overridden. Also, since label is optional it would be good to set a default for the item noun in `addCreateOption`, (like "item")

Reply:
good call!